src/test/java/org/mockito/internal/matchers/CapturingMatcherTest.java

```diff
@@ -52,4 +54,20 @@ public void should_scream_when_nothing_yet_captured() throws Exception {
             fail();
         } catch (MockitoException e) {}
     }
+
+    @Test
+    public void should_not_fail_when_used_in_concurrent_tests() throws Exception {
+        //given
+        final CapturingMatcher<String> m = new CapturingMatcher<String>();
+
+        //when
+        m.captureFrom("concurrent access");
+        Iterator<String> iterator = m.getAllValues().iterator();
+        m.captureFrom("concurrent access");
+
+        //then
+        assertThat(iterator.hasNext()).isTrue();
+        assertThat(iterator.next()).isEqualTo("concurrent access"); // Potential ConcurrentModificationException
```

Comment:
I thins particular test that I added - yes. It simulates the case of unsafe publication of internal storage when interaction with Captor is not finished yet. Here I extracted internal state and created iterator over it. Next captured value will mark iterator as dirty and calling `next()` method on it will cause ConcurrentModificationException.

So in unchanged code it will fail in 100% of the times.

But in my tests I face other errors as well, like NullPointerException on `getLastItem()` but you're right, it's very hard to reproduce it in test.